libraries/Wire/src/Wire.cpp

```diff
@@ -59,192 +14,133 @@ TwoWire::TwoWire(XMC_I2C_t *conf) {
     slaveAddress = 0;
     txAddress = 0;
 
-    rxBufferIndex = 0;
-    rxBufferLength = 0;
-    txBufferIndex = 0;
-    txBufferLength = 0;
-    pre_rxBufferCount = 0;
+    rx_ringBuffer.clear();
+    tx_ringBuffer.clear();
+    pre_rx_ringBuffer.clear();
 }
 
-// Public Methods //////////////////////////////////////////////////////////////
-
 void TwoWire::begin(void) {
-    // Check if desire USIC channel is already in use
-    if ((XMC_I2C_config->channel->CCR & USIC_CH_CCR_MODE_Msk) == XMC_USIC_CH_OPERATING_MODE_SPI) {
-        SPI.end();
-    }
-
+    // To-Do for future work need to check
```

Comment:
Feels like some workaround for some external application which was not calling SPI.end() before using the same pins as SPI. 
If we don´t have the concreate reasons why this was done, this can be removed. 
The user is responsible of calling SPI.end() before using I2C on the same pins.